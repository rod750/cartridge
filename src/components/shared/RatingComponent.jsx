import React from 'react';
import { Icon } from 'bloomer';

export class RatingComponent extends React.Component {
    constructor(props) {
        if(props.rating > 5 || props.rating < 0) {
            throw new Error("Rating prop value can't be more than 5 or less than 0");
        }
        super(props);
    }


    render() {
        return (
            <span>
                {this.renderStars()}
            </span>
        );
    }

    renderStars() {
        var stars = [];
        var count = 0;
        for(var i = 0; i < this.props.rating; i++) {
            stars.push(<Icon className="mdi mdi-star" key={count++} />);
        }
        for(var j = 0; j < 5-this.props.rating; j++) {
            stars.push(<Icon className="mdi mdi-star-outline" key={count++} />);
        }

        return stars;
    }
}
