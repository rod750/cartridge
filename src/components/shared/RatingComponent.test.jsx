import React from 'react';
import ReactDOM from 'react-dom';
import { RatingComponent } from './RatingComponent';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<RatingComponent rating={3} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('it should raise error when rating is more than 5', () => {
    expect(() => {
        renderer.create(<RatingComponent rating={6} />);
    }).toThrow(Error);
});

it('it should raise error when rating is less than 0', () => {
    expect(() => {
        renderer.create(<RatingComponent rating={-1} />);
    }).toThrow(Error);
});

it('it render stars correctly', () => {
    const element = renderer.create(<RatingComponent rating={4} />).toJSON();

    expect(element).toMatchSnapshot();
});

it('it render stars correctly', () => {
    const element = renderer.create(<RatingComponent rating={0} />).toJSON();

    expect(element).toMatchSnapshot();
});

it('it render stars correctly', () => {
    const element = renderer.create(<RatingComponent rating={5} />).toJSON();

    expect(element).toMatchSnapshot();
});
