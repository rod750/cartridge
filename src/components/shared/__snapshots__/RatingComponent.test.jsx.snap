// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it render stars correctly 1`] = `
<span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star"
    />
  </span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star"
    />
  </span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star"
    />
  </span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star"
    />
  </span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star-outline"
    />
  </span>
</span>
`;

exports[`it render stars correctly 2`] = `
<span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star-outline"
    />
  </span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star-outline"
    />
  </span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star-outline"
    />
  </span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star-outline"
    />
  </span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star-outline"
    />
  </span>
</span>
`;

exports[`it render stars correctly 3`] = `
<span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star"
    />
  </span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star"
    />
  </span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star"
    />
  </span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star"
    />
  </span>
  <span
    className="icon"
  >
    <i
      aria-hidden="true"
      className="mdi mdi-star"
    />
  </span>
</span>
`;
